src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs

```diff
@@ -6,27 +6,19 @@
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
 using System.IO;
-using System.Linq;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.Shared.Collections;
 using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
-using Microsoft.VisualStudio.LanguageServices.Implementation.TaskList;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation;
 
 [Export(typeof(AnalyzerFileWatcherService))]
-internal sealed class AnalyzerFileWatcherService
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class AnalyzerFileWatcherService(SVsServiceProvider serviceProvider)
 {
-    private static readonly object s_analyzerChangedErrorId = new();
-
-    private readonly VisualStudioWorkspaceImpl _workspace;
-    private readonly HostDiagnosticUpdateSource _updateSource;
-    private readonly IVsFileChangeEx _fileChangeService;
+    private readonly IVsFileChangeEx _fileChangeService = (IVsFileChangeEx)serviceProvider.GetService(typeof(SVsFileChangeEx));
```

Comment:
Ouch. This is not valid in a MEF part. Will this be removed soon or should I fix it after this change merges?

Reply:
feel free to fix.